tt_metal/impl/buffers/buffer.hpp

```diff
@@ -58,6 +152,124 @@ class Buffer {
 
     uint64_t page_address(uint32_t bank_id, uint32_t page_index) const;
 
+    uint64_t core_address(uint32_t core_id) const;
+
+    CoreRangeSet shard_grid() const {
+        TT_ASSERT(is_sharded(this->buffer_layout_) , "Buffer not sharded");
```

Comment:
these `TT_ASSERT` calls wouldn't be needed if this function existed in a separate `ShardedBuffer` class